sql/date-and-time-types.md

```diff
@@ -0,0 +1,236 @@
+---
+title: date and time types
+category: user guide
```

Comment:
Please refer to other English documents and add a **summary** after the title line.